src/scaladoc/scala/tools/nsc/doc/html/Doclet.scala

```diff
@@ -7,14 +7,19 @@ package scala.tools.nsc
 package doc
 package html
 
+import scala.reflect.internal.Reporter
 import doclet._
 
 /** The default doclet used by the scaladoc command line tool
   * when no user-provided doclet is provided. */
-class Doclet extends Generator with Universer {
+class Doclet(reporter: Reporter) extends Generator with Universer {
 
-  def generateImpl() {
-    new html.HtmlFactory(universe, new ScalaDocReporter(universe.settings)).generate()
-  }
+  @deprecated("Doclets should be created with the Reporter constructor. Otherwise logging reporters will not be shared by the creating parent", "2.12.0")
+  def this() { this(null) }
```

Comment:
It's seems more natural to me to supply the default reporter in the ctor, rather than checking for null in the method.

Also, `def this() = this(new ScalaDocReporter)` looks better.